src/ModbusDevice.cpp

```diff
@@ -30,6 +30,20 @@ ModbusDevice::ModbusDevice(const std::string& name, int8_t slaveAddress)
 {
 }
 
+ModbusDevice::ModbusDevice(const ModbusDevice& device)
+: m_name(device.m_name)
+, m_status(device.m_status)
+, m_groups()
+, m_reader(device.m_reader)
+, m_onMappingValueChange(device.m_onMappingValueChange)
+, m_onStatusChange(device.m_onStatusChange)
```

Comment:
What about slave address?

Reply:
Other device will use a different slave address for sure, so it shouldn't be copied.